test/unit/test_annotated_code.c

```diff
@@ -0,0 +1,298 @@
+	r_vector_push (&test_annotations, &annotation);
+	annotation = makeRCodeAnnotation(11, 15, R_CODE_ANNOTATION_TYPE_OFFSET, 123, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation);
+	annotation = makeRCodeAnnotation(10, 16, R_CODE_ANNOTATION_TYPE_OFFSET, 123, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation);
+	annotation = makeRCodeAnnotation(17, 20, R_CODE_ANNOTATION_TYPE_OFFSET, 32, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation);
+
+	return test_annotations;
+}
+
+static RVector getSomeAnnotationsHelloWorld(){
+	RVector test_annotations;
+	RCodeAnnotation annotation;
+	r_vector_init (&test_annotations, sizeof(RCodeAnnotation), NULL, NULL);
+	//Code Annotations for a hello world program
+	annotation = makeRCodeAnnotation(1, 5, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_DATATYPE);
+	r_vector_push (&test_annotations, &annotation); //1
+	annotation = makeRCodeAnnotation(6, 10, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_NAME);
+	r_vector_push (&test_annotations, &annotation); //2
+	annotation = makeRCodeAnnotation(11, 15, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation); //3
+	annotation = makeRCodeAnnotation(23, 35, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_NAME);
+	r_vector_push (&test_annotations, &annotation); //4
+	annotation = makeRCodeAnnotation(36, 51, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_CONSTANT_VARIABLE);
+	r_vector_push (&test_annotations, &annotation); //5
+	annotation = makeRCodeAnnotation(23, 52, R_CODE_ANNOTATION_TYPE_OFFSET, 4440, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation); //6
+	annotation = makeRCodeAnnotation(58, 64, R_CODE_ANNOTATION_TYPE_OFFSET, 4447, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation); //7
+	annotation = makeRCodeAnnotation(58, 64, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation); //8
+	annotation = makeRCodeAnnotation(58, 64, R_CODE_ANNOTATION_TYPE_OFFSET, 4447, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation); //9	
+	
+	return test_annotations;
+}
+
+
+static RAnnotatedCode *getHelloWorld(){
+	char *test_string = malloc(71 * sizeof(char));
+	strcpy(test_string, "\nvoid main(void)\n{\n    sym.imp.puts(\"Hello, World!\");\n    return;\n}\n");
+	RAnnotatedCode *code = r_annotated_code_new(test_string);
+
+	RVector/*<RCodeAnnotation>*/ test_annotations;
+	test_annotations = getSomeAnnotationsHelloWorld();
+	RCodeAnnotation *annotation;
+	r_vector_foreach (&test_annotations, annotation) {
+		r_annotated_code_add_annotation(code, annotation);
+	}
+	return code;
+}
+
+
+static bool test_r_annotated_code_new(){
```

Comment:
```suggestion
static bool test_r_annotated_code_new() {
```
and unnecessary newline below